Source/EasyNetQ/Producer/DefaultPubSub.cs

```diff
@@ -37,31 +37,32 @@ IAdvancedBus advancedBus
             this.advancedBus = advancedBus;
         }
 
+        /// <inheritdoc />
         public virtual async Task PublishAsync<T>(T message, Action<IPublishConfiguration> configure, CancellationToken cancellationToken)
         {
             Preconditions.CheckNotNull(message, "message");
             Preconditions.CheckNotNull(configure, "configure");
 
+            var cts = CreateCancellationTokenSource(cancellationToken);
```

Comment:
I'm not sure if this method is worth using here and in subscribe. What is `ConnectionConfiguration.Timeout` exactly? 

> Operation timeout seconds. (default is 10)

This says not too much about its meaning. What operations? Consider such usage:
```c#
await PublishAsync(x, a => {}, new CancellationTokenSource().CancelAfter(TimeSpan.FromMinute(1)));
```
Should this call be canceled after one minute or 10 seconds (default value from ConnectionConfiguration sources)?

Reply:
```suggestion
            using var cts = CreateCancellationTokenSource(cancellationToken);
```

Reply:
>This says not too much about its meaning. What operations? Consider such usage:

All operations. 

>Should this call be canceled after one minute or 10 seconds (default value from ConnectionConfiguration sources)?

10 seconds. If you would like to manage timeout by yourself you should disable default timeout by settings `Timeout.InfitineTimeSpan` to `ConnectionConfiguration.Timeout`(or by settings zero in the connection string).

The same approach is used in HttpClient and we agreed about this approach in Slack as well.

Reply:
OK. Just to clarify: `InfiniteTimeSpan` is `new TimeSpan(0, 0, 0, 0, -1)`. What do you mean _zero in the connection string_? `TimeSpan.Zero` ? These are different values.

Reply:
I mean that historically we consider `timeout=0` as an infinite timeout.

Reply:
OK. Is it possible to pass -1 in the connection string? I just thought and wanted to clarify about whether two values 0 and -1 would set the same infinite timeout.

Reply:
Sure. Will do it.

Reply:
Done.